### DISCUSSION TITLE
Error during deployment

### DISCUSSION THREAD

**Discussion by van-vince** [refinedev/refine#5349] [Q&A] [answered]
This works perfectly in development but gave this error  when I was deploying my app
![Screenshot 2023-12-05 at 5 46 59 AM](https://github.com/refinedev/refine/assets/67305320/40ac8241-f50a-419e-8c10-3af2f351b747)

`
      import { HttpError, useUpdate } from "@refinedev/core";
      import { SaveButton, useAutocomplete } from "@refinedev/mui";
      import Autocomplete from "@mui/material/Autocomplete";
      import Box from "@mui/material/Box";
      import Button from "@mui/material/Button";
      import Dialog from "@mui/material/Dialog";
      import DialogActions from "@mui/material/DialogActions";
      import DialogContent from "@mui/material/DialogContent";
      import DialogTitle from "@mui/material/DialogTitle";
      import TextField from "@mui/material/TextField";
      import { UseModalFormReturnType } from "@refinedev/react-hook-form";
      import { Controller, FieldValues, useForm } from "react-hook-form";
      import { ICategory, IOrder, Nullable } from "../../interfaces";
      import {
        FormControl,
        FormHelperText,
        FormLabel,
        MenuItem,
        OutlinedInput,
        Select,
        Stack,
      } from "@mui/material";
      import InputMask from "react-input-mask";
      import React, { useState } from "react";
      import { useLoadScript } from "@react-google-maps/api";
      import usePlacesAutocomplete, {
        getGeocode,
        getLatLng,
      } from "use-places-autocomplete";
      
      export const CreateOrder: React.FC<
        UseModalFormReturnType<IOrder, HttpError, Nullable<IOrder>>
      > = ({
        saveButtonProps,
        refineCore: { onFinish, queryResult, formLoading },
        modal: { visible, close, title },
        register,
        handleSubmit,
        // setValue,
        control,
        formState: { errors },
      }) => {
        const { mutate } = useUpdate();
      
        const {
          ready,
          value,
          setValue,
          suggestions: { status, data },
          clearSuggestions,
        } = usePlacesAutocomplete({
          requestOptions: {
            region: "gh"
          },
        });
        const { isLoaded } = useLoadScript({
          googleMapsApiKey: import.meta.env.VITE_APP_GOOGLE_API_KEY,
          libraries: ["places"],
        });
        if (!isLoaded) return
        // console.log(value);
        return (
          <Dialog
            open={visible}
            onClose={close}
            PaperProps={{ sx: { minWidth: 500 } }}
          >
            <DialogTitle>Create a new order</DialogTitle>
            <DialogContent>
              {/* <form onSubmit={handleSubmit(onFinish)}> */}
              <Stack gap="10px" marginTop="10px">
                <FormLabel required>Pick Up Address</FormLabel>
                <FormControl>
                  <OutlinedInput
                    id="name"
                    placeholder="Name"
                    {...register("storeInfo.name", { required: true })}
                    style={{ height: "40px" }}
                  />
                  {errors.name && (
                    <FormHelperText error>{errors.name.message}</FormHelperText>
                  )}
                </FormControl>
      
                <FormControl>
                  <InputMask
                    mask="(999) 999999999"
                    disabled={false}
                    {...register("storeInfo.contact", {
                      required: true,
                    })}
                  >
                    {/* @ts-ignore */}
                    {(props: TextFieldProps) => (
                      <TextField
                        {...props}
                        size="small"
                        placeholder="Contact"
                        margin="none"
                        variant="outlined"
                      />
                    )}
                  </InputMask>
                  {errors.contact && (
                    <FormHelperText error>
                      {/* @ts-ignore */}
                      {errors.contact.message}
                    </FormHelperText>
                  )}
                </FormControl>
      
                <FormControl>
                  <Autocomplete
                    disabled={!ready}
                    options={data.map((place_id) =>  place_id.description)}
                    renderInput={(params) => (
                      <TextField
                        {...params}
                        size="small"
                        {...register("storeInfo.location")}
                        placeholder="Location"
                        value={value}
                        onChange={(e) => setValue(e.target.value)}
                        InputProps={{
                          ...params.InputProps,
                          type: "search",
                        }}
                      />
                    )}
                  />
                </FormControl>
              </Stack>
      
              <Stack gap="10px" marginTop="50px">
                <FormLabel required>Drop off Address</FormLabel>
                <FormControl>
                  <OutlinedInput
                    id="name"
                    placeholder="Name"
                    {...register("customerInfo.name", { required: true })}
                    style={{ height: "40px" }}
                  />
                  {errors.name && (
                    <FormHelperText error>{errors.name.message}</FormHelperText>
                  )}
                </FormControl>
      
                <FormControl>
                  <InputMask
                    mask="(999) 999999999"
                    disabled={false}
                    {...register("customerInfo.contact", {
                      required: true,
                    })}
                  >
                    {/* @ts-ignore */}
                    {(props: TextFieldProps) => (
                      <TextField
                        {...props}
                        size="small"
                        placeholder="Contact"
                        margin="none"
                        variant="outlined"
                      />
                    )}
                  </InputMask>
                  {errors.contact && (
                    <FormHelperText error>
                      {/* @ts-ignore */}
                      {errors.contact.message}
                    </FormHelperText>
                  )}
                </FormControl>
      
                <FormControl>
                  <Autocomplete
                    disabled={!ready}
                    options={data.map((place_id) =>  place_id.description)}
                    renderInput={(params) => (
                      <TextField
                        {...params}
                        size="small"
                        {...register("customerInfo.location")}
                        placeholder="Location"
                        value={value}
                        onChange={(e) => setValue(e.target.value)}
                        InputProps={{
                          ...params.InputProps,
                          type: "search",
                        }}
                      />
                    )}
                  />
                </FormControl>
              </Stack>
      
              <Stack gap="10px" marginTop="30px">
                <Controller
                  control={control}
                  //@ts-ignore
                  name="paymentType"
                  render={({ field }: { field: any }) => (
                    <FormControl margin="normal" size="small">
                      <FormLabel>Payment Type</FormLabel>
                      <Select
                        defaultValue="COD"
                        {...field}
                        {...register("paymentType", { required: true })}
                        labelId="payment-select"
                      >
                        <MenuItem value="COD">COD</MenuItem>
                        <MenuItem value="Card">Card</MenuItem>
                      </Select>
                    </FormControl>
                  )}
                />
                <FormControl>
                  <OutlinedInput
                    id="item"
                    placeholder="Item Type"
                    {...register("itemType", { required: true })}
                    style={{ height: "40px" }}
                  />
                  {errors.item && (
                    <FormHelperText error>{errors.item.message}</FormHelperText>
                  )}
                </FormControl>
                <FormControl>
                  <OutlinedInput
                    id="price"
                    placeholder="Price"
                    type="Number"
                    {...register("price", { required: true })}
                    style={{ height: "40px" }}
                  />
                  {errors.price && (
                    <FormHelperText error>{errors.price.message}</FormHelperText>
                  )}
                </FormControl>
              </Stack>
              {/* </form> */}
            </DialogContent>
            <DialogActions>
              <Button onClick={close}>Cancel</Button>
              <SaveButton variant="outlined" {...saveButtonProps} />
            </DialogActions>
          </Dialog>
        );
      };
`

**Comment by BatuhanW** [comment, marked as answer]
Hello @van-vince, it's hard to judge but, I noticed this condition:

` if (!isLoaded) return`

So, here you are returning `undefined`. This could be the problem. Try to return an empty element or `null`, that should fix your issue.
